agent_plugin_builder/build_plugin.py

```diff
@@ -0,0 +1,252 @@
+        for os_type in agent_plugin_manifest.supported_operating_systems:
+            generate_vendor_dirs(build_dir_path, os_type, UID, GID)
+    else:
+        if len(agent_plugin_manifest.supported_operating_systems) > 1:
+            common_dir_possible = check_if_common_vendor_dir_possible(build_dir_path, UID, GID)
+            if common_dir_possible:
+                generate_common_vendor_dir(build_dir_path, UID, GID)
+            else:
+                for os_type in agent_plugin_manifest.supported_operating_systems:
+                    generate_vendor_dirs(build_dir_path, os_type, UID, GID)
+
+
+def generate_plugin_config_schema(build_dir_path: Path, agent_plugin_manifest: AgentPluginManifest):
+    """
+    Generate the config-schema file for the plugin. The schema is generated
+    based on the plugin's options model.
+
+    :param build_dir_path: Path to the build directory.
+    :param agent_plugin_manifest: Agent Plugin manifest.
+    """
+    plugin_options_file_path = (
+        build_dir_path / "src" / f"{agent_plugin_manifest.name.lower()}_options.py"
+    )
+    plugin_config_schema_file_path = build_dir_path / CONFIG_SCHEMA
+    plugin_options_model_name = f"{agent_plugin_manifest.name}Options"
+
+    if plugin_config_schema_file_path.exists():
+        logger.info(f"Skipping generating config-schema. Reason: {CONFIG_SCHEMA} already exists")
+        return
+
+    config_schema = {"type": "object"}
+    if plugin_options_file_path.exists():
+        plugin_options_filename = plugin_options_file_path.stem
+        import sys
+
+        sys.path.append(str(build_dir_path / "src"))
+        options = getattr(import_module(plugin_options_filename), plugin_options_model_name)
+        config_schema = {"properties": options.model_json_schema()["properties"]}
+
+    logger.info(f"Generating config-schema for plugin: {agent_plugin_manifest.name}")
+    schema_contents = json.dumps(config_schema)
+    with plugin_config_schema_file_path.open("w") as f:
+        f.write(schema_contents)
+
+
+def create_source_archive(build_dir_path: Path):
+    """
+    Create the source archive for the plugin.
+
+    :param build_dir_path: Path to the build directory.
+    """
+    source_archive = build_dir_path / f"{SOURCE}.tar.gz"
+    source_build_dir_path = build_dir_path / "src"
+
+    logger.info(f"Creating source archive: {source_archive} ")
+    with tarfile.open(str(source_archive), "w:gz") as tar:
+        for item in source_build_dir_path.iterdir():
+            tar.add(item, arcname=item.name, filter=_source_archive_filter)
+
+
+def _source_archive_filter(file_info: tarfile.TarInfo) -> tarfile.TarInfo | None:
+    EXCLUDE_FILES = [
+        "__pycache__",
+        ".mypy_cache",
+        ".pytest_cache",
+        ".git",
+        ".gitignore",
+        ".DS_Store",
+    ]
+
+    if any(exclude in file_info.name for exclude in EXCLUDE_FILES):
+        return None
+    return file_info
+
+
+def create_plugin_archive(
```

Comment:
In my experience, it's common for _all_ of the build artifacts to be created in the `build/` directory and the artifacts for distribution are _copied_ to the `dist/` directory after the build completes.